list-1.py

```diff
@@ -27,7 +27,7 @@ def common_end(a, b):
   element or they have the same last element. Both arrays will be length 1
   or more.
   """
-  return a[0] == b[0] or a[-1] == b[-1]
+  return (len(a) and len(b))>0 and a[0] == b[0] or a[-1] == b[-1]
```

Comment:
Thanks for submitting a PR, I wrote these a long long time ago so I bet there's mistakes and un-pythonic idioms here and there.

The problem already states both arrays will contain one or more items, so the check added is not necessary.

More importantly, your addition _works_ but it's not logically sound and depends on int-to-bool casting logic. When you use `and` on an `int` (what `len` returns), you are casting said it to a boolean (which will always be `True`, unless the number is 0, which won't happen in this case). So you are basically doing `(True and True) > 0`, which will always be true. You'd also need to put the second part of the conditions (`a[0] == b[0] or a[-1] == b[-1]`) in between parenthesis in order to get the `and` operator you added to apply to both.

For a more reasonable check, you could consider the immediate: `(len(a) > 0 and len(b) > 0)`, or something more convoluted like `min([len(a), len(b)]) > 0`

So: `return (len(a) > 0 and len(b) > 0) and (a[0] == b[0] or a[-1] == b[-1])` would work great if we needed to check for array length.

Reply:
Oh great explanation sir, I knew my mistakes